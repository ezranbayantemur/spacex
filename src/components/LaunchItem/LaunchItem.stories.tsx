import LaunchItem from './LaunchItem';
import type {Meta, StoryObj} from '@storybook/react-native';
import {LaunchItemProps} from './LaunchItem.types';

const LaunchItemMeta: Meta<LaunchItemProps> = {
  title: 'LaunchItem',
  component: LaunchItem,
  args: {
    name: 'FalconSat',
    date: '2006-03-17T00:00:00.000Z',
    image: 'https://images2.imgbox.com/94/f2/NN6Ph45r_o.png',
  },
};

export default LaunchItemMeta;

export const Basic: StoryObj<LaunchItemProps> = {};
