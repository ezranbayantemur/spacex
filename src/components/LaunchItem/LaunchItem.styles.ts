import {StyleSheet} from 'react-native';
import {color, font, padding, typography} from '@style';

export default StyleSheet.create({
  container: {
    margin: padding.MEDIUM,
    padding: padding.MEDIUM,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  image: {
    width: 50,
    height: 50,
  },
  name: {
    fontFamily: font.BOLD,
    fontSize: typography.BIG,
  },
  date: {
    fontFamily: font.REGULAR,
    fontSize: typography.MEDIUM,
  },
});
