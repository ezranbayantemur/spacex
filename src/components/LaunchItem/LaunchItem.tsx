import React from 'react';
import {Image, Text, View} from 'react-native';
import {format} from 'date-fns';
import styles from './LaunchItem.styles';
import type {LaunchItemProps} from './LaunchItem.types';

const LaunchItem = ({name, image, date}: LaunchItemProps) => {
  const formattedDate = format(new Date(date), 'dd MMMM yyyy');

  return (
    <View style={styles.container}>
      <Image style={styles.image} resizeMode="contain" source={{uri: image}} />
      <Text style={styles.name}>{name}</Text>
      <Text style={styles.date}>{formattedDate}</Text>
    </View>
  );
};

export default LaunchItem;
