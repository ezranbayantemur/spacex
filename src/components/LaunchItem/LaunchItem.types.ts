export interface LaunchItemProps {
  image: string;
  name: string;
  date: string;
}
