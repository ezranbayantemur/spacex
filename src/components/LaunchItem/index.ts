export * from './LaunchItem.types';

export {default} from './LaunchItem';
