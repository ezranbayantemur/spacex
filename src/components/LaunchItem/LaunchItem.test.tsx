import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import LaunchItem from './LaunchItem';

let wrapper: ReturnType<typeof render>;

describe('Button unit tests', () => {
  beforeEach(() => {
    wrapper = render(
      <LaunchItem
        name={'FalconSat'}
        date={'2006-03-17T00:00:00.000Z'}
        image={'https://images2.imgbox.com/94/f2/NN6Ph45r_o.png'}
      />,
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render props correctly', () => {
    expect(wrapper.queryByText('FalconSat')).not.toBeNull();
    expect(wrapper.queryByText('17 March 2006')).not.toBeNull();
  });
});
