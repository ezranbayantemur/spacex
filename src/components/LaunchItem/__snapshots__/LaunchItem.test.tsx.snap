// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button unit tests should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 16,
      "padding": 16,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "https://images2.imgbox.com/94/f2/NN6Ph45r_o.png",
      }
    }
    style={
      {
        "height": 50,
        "width": 50,
      }
    }
  />
  <Text
    style={
      {
        "fontFamily": "Quicksand-Bold",
        "fontSize": 20,
      }
    }
  >
    FalconSat
  </Text>
  <Text
    style={
      {
        "fontFamily": "Quicksand-Regular",
        "fontSize": 17,
      }
    }
  >
    17 March 2006
  </Text>
</View>
`;
